packages/ruleset/src/functions/pagination-style.js

```diff
@@ -0,0 +1,363 @@
+const { isSdkExcluded, mergeAllOfSchemaProperties } = require('../utils');
+
+module.exports = function(pathObj, _opts, { path }) {
+  return paginationStyle(pathObj, path);
+};
+
+// Rudimentary debug logging that is useful in debugging this rule.
+const debugEnabled = false;
+function debug(msg) {
+  if (debugEnabled) {
+    console.log(msg);
+  }
+}
+
+/**
+ * This function implements the pagination-style rule which performs numerous checks
+ * on a "get" operation that is recognized as a paginated list operation.
+ * An operation is considered to be a paginated "list"-type operation if:
+ * 1. The path doesn't end in a path param reference (e.g. /v1/drinks vs /v1/drinks/{drink_id})
+ * 2. The operation is a "get"
+ * 3. The operation is not excluded via the x-sdk-exclude extension.
+ * 4. The operation's response schema is an object containing an array property.
+ * 5. The operation defines either an "offset" query param or a query param
+ *    whose name is in ["start", "token", "cursor", "page_token"].
+ *
+ * The specific checks that are performed are identified in the comments below (e.g. "Check #1").
+ * Along with a description of each check, you'll find links to the API Handbook related to the check.
+ * Also, there are comments in the corresponding testcase that refer to the various checks
+ * to make it easy to cross-references.
+ *
+ * @param {*} pathItem the path item that potentially contains a paginated "get" operation.
+ * @param {*} path the array of path segments indicating the "location" of the pathItem within the API definition
+ * @returns an array containing the violations found or [] if no violations
+ */
+function paginationStyle(pathItem, path) {
+  debug('>>> Visiting: ' + path.join('.'));
+
+  // The actual path string (e.g. '/v1/resources') will be the last element in 'path'.
+  const pathStr = path[path.length - 1];
+
+  // Retrieve this path item's 'get' operation.
+  const operation = pathItem.get;
+
+  // We'll bail out now if any of the following are true:
+  // 1. If the path string ends with a path param reference (e.g. '{resource_id}'
+  // 2. If the path item doesn't have a 'get' operation.
+  // 3. The 'get' operation is excluded via the 'x-sdk-exclude' extension.
```

Comment:
I'm not sure I see justification for skipping based on SDK exclusion.

Reply:
My thinking was that since this new rule is a bit stricter than the old rule, users are likely to see new warnings crop up since very few APIs actually follow the letter of the law wrt pagination.  With that in mind, it might be a little overbearing if we start returning a bunch of new warnings for an operation that is excluded from SDK generation.